### PR TITLE
core: move to Hyprutils::OS::CProcess for spawning processes

### LINKED ISSUE #574: Hyprlock does not free memory
### Regression?

Yes

### Hyprlock Info and Version

<Enter your hyprlock version here>
<details> <summary>Hyprlock config</summary>

```sh
#  _                      _            _    
# | |__  _   _ _ __  _ __| | ___   ___| | __
# | '_ \| | | | '_ \| '__| |/ _ \ / __| |/ /
# | | | | |_| | |_) | |  | | (_) | (__|   < 
# |_| |_|\__, | .__/|_|  |_|\___/ \___|_|\_\
#        |___/|_|                           
# 

general {
    ignore_empty_input = true
    grace = 0
    disable_loading_bar = true
}

background {
    monitor =
    # path = $HOME/.config/ml4w/cache/blurred_wallpaper.png   # only png supported for now
    # path = $HOME/Downloads/Wallpapers/blade-runner-3.jpg
    # path = $HOME/Downloads/Wallpapers/wallhaven-gpx537_1920x1080.png
    # path = /home/mellow/Downloads/Wallpapers/2024-05-15.png
    # path = /home/mellow/Downloads/Wallpapers/blade-runner-3.jpg
    path = /home/mellow/Downloads/Wallpapers/wallhaven-q67z9d_1920x1080.png
    blur_passes = 1
    blur_size = 8
    contrast = 1
    noise = 0
    brightness = 0.5
    vibrancy = 0
}

input-field {
    monitor =
    # size = 100, 25
    size = 250, 60
    outline_thickness = 2
    # dots_size = 0.33 # Scale of input-field height, 0.2 - 0.8
    dots_size = 0.2 # Scale of input-field height, 0.2 - 0.8
    dots_spacing = 0.35 #0.15 # Scale of dots' absolute size, 0.0 - 1.0
    dots_center = true
    dots_rounding = -1 # -1 default circle, -2 follow input-field rounding
    outer_color = rgb(151515)
    inner_color = rgb(FFFFFF)
    font_color = rgb(10, 10, 10)
    fade_on_empty = true
    fade_timeout = 1000 # Milliseconds before fade_on_empty is triggered.
    placeholder_text = <i>Enter Password...</i> # Text rendered in the input box when it's empty.
    hide_input = false
    rounding = -1 # -1 means complete rounding (circle/oval)
    check_color = rgb(204, 136, 34)
    fail_color = rgb(204, 34, 34) # if authentication failed, changes outer_color and fail message color
    fail_text = <i>$FAIL <b>($ATTEMPTS)</b></i> # can be set to empty
    fail_transition = 300 # transition time in ms between normal outer_color and fail_color
    capslock_color = -1
    numlock_color = -1
    bothlock_color = -1 # when both locks are active. -1 means don't change outer color (same for above)
    invert_numlock = false # change color if numlock is off
    swap_font_color = false # see below
    position = 0, -200 #30, 50
    halign = center #left
    valign = center #bottom
}

label {
    monitor =
    #clock
    # text = cmd[update:1000] echo $TIME12
    text = cmd[update:1000] date +"%I:%M %p"
    color = rgba(200, 200, 200, 1.0)
    font_size = 70
    font_family = JetBrains Mono Nerd Font Mono Bold
    position = 0, 100 #-100, 70
    halign = center
    valign = center
    shadow_passes = 5
    shadow_size = 10
}

label {
    text = $USER
    color = rgba(200, 200, 200, 1.0)
    font_size = 30
    font_family = Iosevka
    position = 0, 400
    halign = center
    valign = bottom
    shadow_passes = 5
    shadow_size = 10
}

image {
    monitor =
    # path = $HOME/.config/ml4w/cache/square_wallpaper.png
    # path =  $HOME/Downloads/Wallpapers/wallhaven-pkvw9p_1920x1080.png
    # path = $HOME/Downloads/Wallpapers/wallhaven-gpx537_1920x1080.png
    size = 280 # lesser side if not 1:1 ratio
    rounding = -1 # negative values mean circle
    border_size = 4
    border_color = rgb(221, 221, 221)
    rotate = 0 # degrees, counter-clockwise
    reload_time = -1 # seconds between reloading, 0 to reload with SIGUSR2
#    reload_cmd =  # command to get new path. if empty, old path will be used. don't run "follow" commands like tail -F
    position = 0, 200
    halign = center
    valign = center
}
```

</details>

### Compositor Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit  dirty ().
Date:
Tag: , commits:
built against aquamarine 0.4.1


flags: (if any)


System Information:
System name: Linux
Node name: mellow
Release: 6.11.5-1-default
Version: #1 SMP PREEMPT_DYNAMIC Wed Oct 23 04:27:11 UTC 2024 (b4e3aa9)


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation UHD Graphics 620 [8086:5917] (rev 07) (prog-if 00 [VGA controller])


os-release: NAME="openSUSE Tumbleweed"
# VERSION="20241025"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20241025"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
# CPE 2.3 format, boo#1217921
CPE_NAME="cpe:2.3:o:opensuse:tumbleweed:20241025:*:*:*:*:*:*:*"
#CPE 2.2 format
#CPE_NAME="cpe:/o:opensuse:tumbleweed:20241025"
BUG_REPORT_URL="https://bugzilla.opensuse.org"
SUPPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"


plugins:

```

</details>

### Description

So, in `btop`, I was checking memory usage and I usually use `hyprlock` and `hypridle` for suspending the system after some idle time and I don't restart the system very much. I saw that these processes are still using memory of my system.

Is this expected to happen? cause  I don't have lots of free memory!

image

### How to reproduce

- Use `hyprlock` with `hypridle` to suspend the system and see the allocated memory from `btop`.

### Crash reports, logs, images, videos

_No response_

### PR DESCRIPTION
**Pull request by PaideiaDilemma**
Only thing is that now stderr is not printed. Maybe we want to print that if it's not empty?

Closes #574

### CODE REVIEW

**Comment by vaxerski** [comment]
whats the problem in grabbing `.stderr()`?

**Comment by PaideiaDilemma** [comment]
Nothing really. This should do it.

File: src/core/hyprlock.cpp [hunk 1]
```diff
@@ -1105,74 +1108,15 @@ void CHyprlock::enqueueForceUpdateTimers() {
...
-    waitpid(child, NULL, 0);
-
-    if (child < 0) {
-        Debug::log(LOG, "Failed to create the second fork");
-        return;
-    }
+std::string CHyprlock::spawnSync(const std::string& cmd) {
+    CProcess proc("/bin/sh", {"-c", cmd});
+    if (!proc.runSync())
+        Debug::log(ERR, "Failed to run \"{}\"", cmd);
```
Comment (vaxerski) [inline src/core/hyprlock.cpp:1114]: you should return early, nothing will be in stdout/stderr

**Review by vaxerski — APPROVED** [review]
livin on a prayer

**Pull request merged**